### PR TITLE
Employee Education Result Feature

### PR DESCRIPTION
**Pull request by ManiruzzamanAkash**
[update] Add Result feature in education tab. Result, out of scale and types(grade/percentage) are added.

### CODE REVIEW

**Review by tanmayjay — APPROVED** [review]

File: includes/updates/update-1.8.3.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            $wpdb->prepare(
+                "ALTER TABLE `{$wpdb->prefix}erp_hr_education` ADD `result` JSON NULL DEFAULT NULL AFTER `field`;"
+            )
+        );
+    }
+
+    if ( ! in_array( 'result_type', $cols ) ) {
+        $wpdb->query(
+            $wpdb->prepare(
+                "ALTER TABLE `{$wpdb->prefix}erp_hr_education` ADD `result_type` ENUM('grade','percentage') NULL DEFAULT NULL AFTER `result`;"
```
Comment (tanmayjay) [inline includes/updates/update-1.8.3.php:22]: Why this file changes exist in #1243 ?
On top of that the definitions are different!

File: modules/hrm/includes/functions-employee.php [hunk 2]
```diff
@@ -1105,3 +1105,27 @@ function get_employee_additional_fields( $fields, $id, $user ) {
...
+    ];
+
+    $types = apply_filters( 'erp_hr_education_result_type_option', $types );
+
+    if ( $selected ) {
+        return ( isset( $types[ $selected ] ) ) ? $types[ $selected ] : '';
+    }
+
+    return $types;
+}
```
Comment (tanmayjay) [inline modules/hrm/includes/functions-employee.php:1131]: Same function in #1243 with completely different definition

File: modules/hrm/views/js-templates/education-form.php [hunk 3]
```diff
@@ -46,6 +46,50 @@
...
+            'required'    => true,
+            'placeholder' => '5.0',
+            'id'          => 'scale',
+            'custom_attr' => [
+                'min'  => 0,
+                'step' => '0.01'
+            ],
+        ] ); ?>
+    </div>
+
```
Comment (tanmayjay) [inline modules/hrm/views/js-templates/education-form.php:92]: Same changes in #1243 with different codes

File: includes/class-install.php [hunk 4]
```diff
@@ -746,6 +746,8 @@ public function create_tables() {
                 `school` varchar(100) DEFAULT NULL,
                 `degree` varchar(100) DEFAULT NULL,
                 `field` varchar(100) DEFAULT NULL,
+                `result` JSON DEFAULT NULL,
+                `result_type` enum('grade', 'percentage') DEFAULT NULL,
```
Comment (tanmayjay) [inline includes/class-install.php:750]: Same thing with different schema exists in #1243

**Review by tanmayjay — CHANGES_REQUESTED** [review]

**Comment by ManiruzzamanAkash** [comment]
Fixed redundant code issue on #1243

**Review by tanmayjay — APPROVED** [review]

**Pull request merged**